store/postgres/src/chain_store.rs

```diff
@@ -1028,6 +1028,69 @@ mod data {
                 .collect())
         }
 
+        pub(super) fn clear_call_cache(
+            &self,
+            conn: &PgConnection,
+            from: Option<i32>,
+            to: Option<i32>,
+        ) -> Result<usize, Error> {
+            match self {
+                Storage::Shared => {
+                    use public::eth_call_cache as cache;
+                    let delete_stmt = diesel::delete(cache::table);
+                    match (from, to) {
+                        (Some(from), None) => delete_stmt
+                            .filter(cache::block_number.ge(from))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (None, Some(to)) => delete_stmt
+                            .filter(cache::block_number.le(to))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (Some(from), Some(to)) => delete_stmt
+                            .filter(cache::block_number.ge(from))
+                            .filter(cache::block_number.le(to))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (None, None) => delete_stmt.execute(conn).map_err(Error::from),
+                    }
+                }
```

Comment:
I'd suggest mutating the `delete_stmt` value only once for each variant, and only calling `.execute()` at the end. 
This reduces code duplication, which helps with maintenance.

Also, since this function's error type is `anyhow::Error`, we can imbue extra details about possible errors. This would also occur only once in this scope.

```suggestion
            use anyhow::Context;
            match self {
                Storage::Shared => {
                    use public::eth_call_cache as cache;
                    let mut delete_stmt = diesel::delete(cache::table);
                    if let Some(from) = from {
                        delete_stmt.filter(cache::block_number.ge(from));
                    }
                    if let Some(to) = from {
                        delete_stmt.filter(cache::block_number.le(to));
                    }
                    delete_stmt.execute(conn).context(
                        "Failed to execute SQL to delete the block cache in shared storage.",
                    )
                }
```

Reply:
`filter` returns `Filter<Self, Predicate>`  instead of mutating `delete_stmt`. I'm worried that, filters won't be applied on `delete_stmt`

Reply:
Good catch! 
In this case, we can use the [`.into_boxed` method](https://docs.rs/diesel/2.0.2/diesel/query_builder/struct.DeleteStatement.html#method.into_boxed) and reassign the new value to the `delete_stmt` variable:
```rust
let mut delete_stmt = diesel::delete(cache::table).into_boxed();
if let Some(from) = from {
    delete_stmt = delete_stmt.filter(cache::block_number.ge(from));
}
if let Some(to) = from {
    delete_stmt = delete_stmt.filter(cache::block_number.le(to));
}
```